StrongInject.Tests.Unit/OwnedInjectionTests.cs

```diff
@@ -0,0 +1,1275 @@
+            comp.GetDiagnostics().Verify();
+            var file = Assert.Single(generated);
+            file.Should().BeIgnoringLineEndings(@"#pragma warning disable CS1998
+partial class Container
+{
+    private int _disposed = 0;
+    private bool Disposed => _disposed != 0;
+    public async global::System.Threading.Tasks.ValueTask DisposeAsync()
+    {
+        var disposed = global::System.Threading.Interlocked.Exchange(ref this._disposed, 1);
+        if (disposed != 0)
+            return;
+    }
+
+    async global::System.Threading.Tasks.ValueTask<TResult> global::StrongInject.IAsyncContainer<global::A>.RunAsync<TResult, TParam>(global::System.Func<global::A, TParam, global::System.Threading.Tasks.ValueTask<TResult>> func, TParam param)
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::StrongInject.AsyncOwned<global::B> asyncOwned_0_1;
+        global::A a_0_0;
+        async global::System.Threading.Tasks.Task<global::StrongInject.AsyncOwned<global::B>> CreateAsyncOwnedB_2()
+        {
+            global::B b_0_0;
+            b_0_0 = new global::B();
+            return new global::StrongInject.AsyncOwned<global::B>(b_0_0, async () =>
+            {
+                await ((global::System.IAsyncDisposable)b_0_0).DisposeAsync();
+            });
+        }
+
+        asyncOwned_0_1 = await CreateAsyncOwnedB_2();
+        a_0_0 = new global::A(b: asyncOwned_0_1);
+        TResult result;
+        try
+        {
+            result = await func(a_0_0, param);
+        }
+        finally
+        {
+        }
+
+        return result;
+    }
+
+    async global::System.Threading.Tasks.ValueTask<global::StrongInject.AsyncOwned<global::A>> global::StrongInject.IAsyncContainer<global::A>.ResolveAsync()
+    {
+        if (Disposed)
+            throw new global::System.ObjectDisposedException(nameof(Container));
+        global::StrongInject.AsyncOwned<global::B> asyncOwned_0_1;
+        global::A a_0_0;
+        async global::System.Threading.Tasks.Task<global::StrongInject.AsyncOwned<global::B>> CreateAsyncOwnedB_2()
+        {
+            global::B b_0_0;
+            b_0_0 = new global::B();
+            return new global::StrongInject.AsyncOwned<global::B>(b_0_0, async () =>
+            {
+                await ((global::System.IAsyncDisposable)b_0_0).DisposeAsync();
+            });
+        }
+
+        asyncOwned_0_1 = await CreateAsyncOwnedB_2();
+        a_0_0 = new global::A(b: asyncOwned_0_1);
+        return new global::StrongInject.AsyncOwned<global::A>(a_0_0, async () =>
+        {
+        });
+    }
+}");
+        }
+
+        [Fact]
+        public void InjectParameterFuncOfOwnedWithAsyncInitializationOfDependency()
+        {
+            string userSource = @"
+using System;
+using System.Threading.Tasks;
+using StrongInject;
+
+[Register(typeof(A)), Register(typeof(B))]
+public partial class Container : IAsyncContainer<A> { }
```

Comment:
This also shouldn't require an AsyncContainer

Reply:
Updated.